src/Common/ProfileEvents.cpp

```diff
@@ -190,8 +190,13 @@
     \
     M(ReplicaPartialShutdown, "How many times Replicated table has to deinitialize its state due to session expiration in ZooKeeper. The state is reinitialized every time when ZooKeeper is available again.") \
     \
+    M(PrimaryKeyNotUsed, "Check if primary key is used in this query.") \
+    M(TotalParts, "Total number of data parts in a MergeTree table.") \
+    M(PartitionParts, "Number of data parts selected by partition in a MergeTree table.") \
     M(SelectedParts, "Number of data parts selected to read from a MergeTree table.") \
     M(SelectedRanges, "Number of (non-adjacent) ranges in all data parts selected to read from a MergeTree table.") \
+    M(TotalMarks, "Total number of marks (index granules) in a MergeTree table.") \
```

Comment:
About TotalMarks and PrimaryKeyMarks: We have a similar problem here as above. Updating statistics while running queries is by definition not able to compute the total number of marks in a MergeTree Table.

Reply:
seems I have some misconceptions about profile events...